odoo/tools/translate.py

```diff
@@ -888,6 +889,45 @@ def handle_text(text, lineno):
     _extract_translatable_qweb_terms(tree.getroot(), handle_text)
     return result
 
+
+def extract_spreadsheet_terms(fileobj, keywords, comment_tags, options):
+    """Babel message extractor for spreadsheet data files.
+
+    :param fileobj: the file-like object the messages should be extracted from
+    :param keywords: a list of keywords (i.e. function names) that should
+                     be recognized as translation functions
+    :param comment_tags: a list of translator tags to search for and
+                         include in the results
+    :param options: a dictionary of additional options (optional)
+    :return: an iterator over ``(lineno, funcname, message, comments)``
+             tuples
+    :rtype: Iterable
+    """
+    terms = []
+    data = json.load(fileobj)
+    for sheet in data.get('sheets', []):
+        for cell in sheet['cells'].values():
+            content = cell.get('content', '')
+            terms += re.findall(r'_t\("(.+?)"\)', content, re.IGNORECASE)
+            markdown_link = re.fullmatch(r'\[(.+)\]\(.+\)', content)
+            if markdown_link:
+                terms.append(markdown_link.group(1))
+        for figure in sheet['figures']:
+            terms.append(figure['data']['title'])
+            if 'baselineDescr' in figure['data']:
+                terms.append(figure['data']['baselineDescr'])
+    pivots = data.get('pivots', dict()).values()
+    lists = data.get('lists', dict()).values()
```

Comment:
```suggestion
    pivots = data.get('pivots', {}).values()
    lists = data.get('lists', {}).values()
```

Reply:
Also `dict.get` will return the default value if the key is missing, is that always the case or is it possible for the key to be present, mapped to e.g. `null`?